app/BUILD.bazel

```diff
@@ -370,8 +374,46 @@ in order to keep Gradle building.
      sed 's/import org.oppia.app.R/import org.oppia.app.databinding.adapters.R/g' |
      sed 's/BindingAdapters {/BindingAdapters_updated {/g' > $(OUTS)
      ''',
+ ) for file in BINDING_ADAPTERS_WITH_RESOURCE_IMPORTS]
+
+'''
+Files to be excluded from the source list for the app library. This list includes all the files
+built by the following libraries:
+- view_models
+- views
+- listeners
+- annotations
+- binding_adapters
+'''
+EXCLUDED_APP_LIB_FILES = VIEW_MODELS + VIEW_MODELS_WITH_RESOURCE_IMPORTS + VIEWS + VIEWS_WITH_RESOURCE_IMPORTS + BINDING_ADAPTERS + BINDING_ADAPTERS_WITH_RESOURCE_IMPORTS + LISTENERS + ANNOTATIONS
+
+'''
+Files to be built by the app library. All of these files are assumed to have resource import and
+will be processed by the app library genrule regardless in order to avoid any import issues.
```

Comment:
```suggestion
Files to be built by the app library. All of these files are assumed to have resource imports and
will be processed by the app library genrule regardless in order to avoid any import issues.
```

Reply:
Done